quartz/plugins/transformers/links.ts

```diff
@@ -55,7 +57,29 @@ export const CrawlLinks: QuartzTransformerPlugin<Partial<Options> | undefined> =
               ) {
                 let dest = node.properties.href as RelativeURL
                 const classes = (node.properties.className ?? []) as string[]
-                classes.push(isAbsoluteUrl(dest) ? "external" : "internal")
+                const isExternal = isAbsoluteUrl(dest)
+                classes.push(isExternal ? "external" : "internal")
+
+                if (isExternal && opts.externalLinkIcon) {
+                  node.children.push({
+                    type: "element",
+                    tagName: "svg",
+                    properties: {
+                      class: "external-icon",
+                      viewBox: "0 0 512 512",
+                    },
+                    children: [
+                      {
+                        type: "element",
+                        tagName: "path",
+                        properties: {
+                          d: "M320 0H288V64h32 82.7L201.4 265.4 178.7 288 224 333.3l22.6-22.6L448 109.3V192v32h64V192 32 0H480 320zM32 32H0V64 480v32H32 456h32V480 352 320H424v32 96H64V96h96 32V32H160 32z",
+                        },
+                        children: [],
+                      },
+                    ],
+                  })
+                }
```

Comment:
this works but I would prefer to do this at a CSS level via something like https://stackoverflow.com/questions/19255296/is-there-a-way-to-use-svg-as-content-in-a-pseudo-element-before-or-after

this means that this feature won't impact build times and page size

Reply:
I don't believe that using the `content: url()` method would work though. In order to achieve color scheme responsiveness, it's necessary for the SVG to use CSS variables. I can't seem to get it to work with the `content: url()` method though, possibly because it's not actually part of the DOM?

Reply:
the svg can just `currentColor` for stroke and then style it separately in `base.scss`, it works I've done it

Reply:
@jackyzha0 not sure what you mean, here's what i'm doing currently (it's not working)
```css
&.external::after {
    color: var(--dark);
    content: url(/static/external-icon.svg);
    vertical-align: bottom;
    height: 1ex;
    margin-left: 0.15em;
  }
```

```xml
<svg viewBox="0 0 512 512">
    <path fill="currentColor" d="M320 0H288V64h32 82.7L201.4 265.4 178.7 288 224 333.3l22.6-22.6L448 109.3V192v32h64V192 32 0H480 320zM32 32H0V64 480v32H32 456h32V480 352 320H424v32 96H64V96h96 32V32H160 32z" />
</svg>
```

Reply:
https://css-tricks.com/lodge/svg/09-svg-data-uris/

Reply:
Not sure how that helps. The problem I think is that the SVG isn't receiving the color defined from `var(--dark)` for some reason. This applies both with data and path URIs.

Reply:
ah I see what you mean, seems like the `svg` is being treated as its own separate document hm, let's go with your original approach